system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -254,45 +254,35 @@ void processHitsJsonResp(IPropertyTreeIterator * iter, StringBuffer & returnbuf,
             ForEach(*iter)
             {
                 IPropertyTree & cur = iter->query();
+                Owned<IPropertyTreeIterator> fieldElementsItr = cur.getElements("*");
+
                 bool first = true;
-                Owned<IPropertyTreeIterator> fieldelementsitr = cur.getElements("*");
-                StringBuffer postheader;
-                ForEach(*fieldelementsitr)
+                if (reportHeader)
                 {
-                    if (reportHeader)
+                    ForEach(*fieldElementsItr)
                     {
                         if (!first)
-                        {
                             returnbuf.append(", ");
-                            postheader.append(", ");
-                        }
                         else
-                        {
                             first = false;
-                        }
-
-                        StringBuffer fieldName;
-                        fieldelementsitr->query().getName(fieldName);
-                        returnbuf.append(fieldName.str());
-
-                        postheader.append(fieldelementsitr->query().queryProp(".")); // commas in data should be escaped
+                        fieldElementsItr->query().getName(returnbuf);
                     }
-                    else
-                    {
-                        if (!first)
-                            returnbuf.append(", ");
-                        else
-                            first = false;
+                    returnbuf.newline();
+                    first = true;
 
-                        returnbuf.append(fieldelementsitr->query().queryProp(".")); // commas in data should be escaped
-                    }
+                    reportHeader = false;
                 }
-                returnbuf.append("\n");
-                if (reportHeader)
+
+                ForEach(*fieldElementsItr)
                 {
-                    returnbuf.appendf("%s\n", postheader.str());
-                    reportHeader = false;
+                    if (!first)
+                        returnbuf.append(", ");
+                    else
+                        first = false;
+
+                    returnbuf.append(fieldElementsItr->query().queryProp(nullptr)); // commas in data should be escaped
```

Comment:
super trivial/don't change unless you want to: to be consistent with other loop could be fieldElementsItr->query().getName(returnbuf) too.

Reply:
the closest function I found to be consistent with the neighboring loop is:
fieldElementsItr->query().getProp(nullptr, returnbuf);

Reply:
I didn't look closely enough with previous comment, but yes, I meant a getXX method instead of the queryXX method, so the method does the appending.

so yes:
```
fieldElementsItr->query().getProp(nullptr, returnbuf); 
```